src/main/java/duke/command/EventCommand.java

```diff
@@ -0,0 +1,58 @@
+package duke.command;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+
+import duke.exception.DukeException;
+import duke.storage.Storage;
+import duke.task.Event;
+import duke.tasklist.TaskList;
+import duke.ui.Ui;
+
+/**
+ * The EventCommand class encapsulates the execution of the event command from the user.
+ */
+public class EventCommand extends Command {
+
+    /** The event command inputted by the user. */
+    private String fullCommand;
+
+    /**
+     * Constructor to intialise an EventCommand.
+     *
+     * @param fullCommand The event command inputted by the user.
+     */
+    public EventCommand(String fullCommand) {
+        this.fullCommand = fullCommand;
+    }
+
+    /**
+     * Executes the response to the event command from the user.
+     *
+     * @param storage The storage Duke uses to save and load the tasklist from.
+     * @param taskList The list of tasks Duke needs to execute on.
+     * @param ui The Ui Duke utilises to interact with the user.
+     */
+    @Override
+    public void execute(Storage storage, TaskList taskList, Ui ui) throws IOException, DukeException {
```

Comment:
The execute in many of the commands does many similar things, and maybe has the responsibility to do too many things; perhaps making the flow more modular could be nice?